python/cudf/cudf/core/scalar.py

```diff
@@ -112,29 +114,44 @@ def _device_value_to_host(self):
         self._host_value = self._device_value._to_host_scalar()
 
     def _preprocess_host_value(self, value, dtype):
+        if isinstance(dtype, Decimal64Dtype):
+            # TODO: Support coercion from decimal.Decimal to different dtype
+            # TODO: Support coercion from integer to Decimal64Dtype
+            raise NotImplementedError(
+                "dtype as cudf.Decimal64Dtype is not supported. Use Decimal "
+                "to construct a DecimalScalar."
```

Comment:
Just having `Decimal` here is a bit unclear that we're referring to a python `decimal.Decimal` object